src/record-matching/bib/generate-query-list.js

```diff
@@ -33,12 +33,15 @@ export function generateIdentifierQueries(record) {
 	function getStandardIdentifiers() {
 		return record.get(/^020|022|024$/)
 			.reduce((acc, field) => {
-				const subfield = field.subfields.find(sf => sf.code === 'a');
+				const codes = ['a', 'z'];
+				field.subfields
+					.filter(sub => {
+						if (codes.includes(sub.code) && !acc.includes(sub.value)) {
+							acc.push(sub.value);
+						}
 
-				if (subfield) {
-					const id = subfield.value;
-					return id in acc ? acc : acc.concat(id);
-				}
+						return null;
```

Comment:
Return `undefined` or `false`. Doesn't really matter here but returning `null` should only be done when it's explicitly needed (`null` is an object and can mess up some comparisons).